### DISCUSSION TITLE
How to generate a bundle analysis

### DISCUSSION THREAD

**Discussion by moeriki** [frandiox/vite-ssr#114] [Q&A] [answered]
Because vite-ssr runs rollup twice any bundle analyser would overwrite the analysis of the first run (client bundle) with the analysis of the second run (SSR bundle).

So far this is the best I have come up with.

```ts
      plugins: [
        visualizer({
          get filename() {
            return `dist/stats_${String(Date.now()).slice(-5)}.html`;
          },
        }),
      ],
```

Is there a way to know which run we're in, in the config of another plugin? Any other idea?

**Comment by frandiox** [comment, marked as answer]
@Moeriki You can pass options only to client or server builds in the plugin:

```js
viteSSR({
      build: {
        clientOptions: {
          plugins: []
        },
        serverOptions: {
          plugins: []
        }
      }
    })
```

If you want to visualize both, I guess you can add the plugin twice with different file output?
